priv/perf/apps/load_test/lib/scenario/utxos.ex

```diff
@@ -0,0 +1,141 @@
+    |> await_all(:create_utxos_and_make_assertions)
+  end
+
+  def create_utxos_and_make_assertions(session) do
+    with {:ok, sender, receiver} <- create_accounts(),
+         {:ok, utxo} <- fund_account(session, sender),
+         :ok <- spend_utxo(session, utxo, sender, receiver) do
+      Session.add_metric(session, "error_rate", 0)
+    else
+      error ->
+        log_error(session, "#{__MODULE__} failed with #{inspect(error)}")
+
+        session
+        |> Session.add_metric("error_rate", 1)
+        |> Session.add_error(:error, error)
+    end
+  end
+
+  defp create_accounts() do
+    {:ok, sender_address} = Account.new()
+    {:ok, receiver_address} = Account.new()
+
+    {:ok, sender_address, receiver_address}
+  end
+
+  defp fund_account(session, account) do
+    initial_amount = config(session, [:chain_config, :initial_amount])
+    token = config(session, [:chain_config, :token])
+
+    with {:ok, utxo} <- fund_childchain_account(account, initial_amount, token),
+         :ok <-
+           fetch_childchain_balance(account,
+             amount: initial_amount,
+             token: Encoding.to_binary(token),
+             error: :wrong_childchain_after_funding
+           ),
+         :ok <- validate_utxos(account, %{utxo | owner: account.addr}) do
+      {:ok, utxo}
+    end
+  end
+
+  defp validate_utxos(account, utxo) do
+    utxo_with_owner =
+      case utxo do
+        :empty -> :empty
+        _ -> %{utxo | owner: account.addr}
+      end
+
+    case Utxo.get_utxos(account, utxo_with_owner) do
+      {:ok, _} -> :ok
+      _other -> :invalid_utxos
+    end
+  end
+
+  defp fund_childchain_account(address, amount, token) do
+    case Faucet.fund_child_chain_account(address, amount, token) do
+      {:ok, utxo} -> {:ok, utxo}
+      _ -> :failed_to_fund_childchain_account
+    end
+  end
+
+  defp spend_utxo(session, utxo, sender, receiver) do
+    amount = config(session, [:chain_config, :initial_amount])
+    token = config(session, [:chain_config, :token])
+    fee = config(session, [:chain_config, :fee])
+    amount_to_transfer = amount - fee
+
+    with [new_utxo] <- Transaction.spend_utxo(utxo, amount_to_transfer, fee, sender, receiver, token),
+         :ok <- validate_utxos(sender, :empty),
+         :ok <- validate_utxos(receiver, %{new_utxo | owner: receiver.addr}),
+         :ok <-
+           fetch_childchain_balance(sender, amount: 0, token: Encoding.to_binary(token), error: :wrong_sender_balance),
+         :ok <-
+           fetch_childchain_balance(receiver,
+             amount: amount_to_transfer,
+             token: Encoding.to_binary(token),
+             error: :wrong_sender_balance
+           ) do
+      :ok
+    end
+  end
+
+  defp fetch_childchain_balance(account, amount: amount, token: token, error: error) do
+    childchain_balance = Balance.fetch_balance(account.addr, amount, token)
+
+    case childchain_balance do
+      nil when amount == 0 -> :ok
+      %{"amount" => ^amount} -> :ok
```

Comment:
are you sure this two are the same?

Reply:
`nil when amount == 0` means we're expecting amount to be empty in the childchain for the specified currency. if it's not found in the json returned from watcherinfo, it means amount is zero.